hotrod/services/frontend/Sources/API/Request.swift

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing Examples open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing Examples project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+import Instrumentation
+import NIOHTTP1
+import Tracing
+
+struct Request: Equatable {
+    let startTime: DispatchTime
+    let head: HTTPRequestHead
+    let span: Span
```

Comment:
are you sure about carrying a span? should this not carry baggage instead?

Reply:
I needed some way of carrying the `Span` to `end` it once the request was complete. Another possibility would be to store the `Span` inside a `State` enum case, similar to how I stored the `Request` in there 🤔

Reply:
Had a call with @ktoso about this. We agreed on splitting up the information stored in `Request` into separate variables stored in the state enum cases.